data_safe_haven/resources/workspace/ansible/desired_state.yaml

```diff
@@ -148,6 +148,53 @@
         creates: /etc/audit/rules.d/50-privileged.rules
       notify: Restart auditd
 
+    - name: Copy ClamAV configuration
+      ansible.builtin.copy:
+        src: etc/clamav/clamd.conf
+        dest: /etc/clamav/clamd.conf
+        mode: '0644'
+
+    # This also generates a default freshclam.conf if one does not exist
+    - name: Enable and start freshclam service
+      ansible.builtin.systemd:
+        name: clamav-freshclam
+        state: started
+        enabled: true
+
+    - name: Set freshclam private mirror
+      ansible.builtin.lineinfile:
+        path: /etc/clamav/freshclam.conf
+        line: "{{ lookup('file', '/etc/clamav/freshclam-mirror.conf') }}"
+        state: present
+      changed_when: true
```

Comment:
```suggestion
```
You shouldn't need that. The module will trigger the handler whenever the file is changed.

Reply:
See notes above. The module *doesn't* trigger the handler when the file is changed without this.

Reply:
I'm about 99.9% certain if the file is updated, the `lineinfile` module will report changes and the handler will be run.

My guess is what happened is the `notify` was added after the file had been edited, so when running the playbook again the module reports OK and the handler isn't called.

Reply:
Using `changed_when: true` or `changed_when: false` is bad practice because tasks will always report as changed or OK regardless of what the module does.

Reply:
As I said above, if you know a better way to trigger the notify if-and-only-if the file is changed, I'm happy to update this.

Reply:
I might not be explaining very well. I'm sure that is exactly what should happen without any `changed_when` declaration. Like we already do for sshd and auditd.

I'll deploy this branch (without the `changed_when`) and check.